backend/src/data/repositories/interview/interview.repository.ts

```diff
@@ -22,7 +23,19 @@ class Interview {
       .query()
       .withGraphJoined('courseCategory')
       .withGraphJoined('interviewee')
+      .modifyGraph('interviewee', (builder) => {
+        builder
+          .select('users.id', 'users.createdAt', 'email', 'fullName')
+          .joinRelated('userDetails')
+          .castTo<UsersGetResponseDto>();
+      })
       .withGraphJoined('interviewer')
+      .modifyGraph('interviewer', (builder) => {
+        builder
+          .select('users.id', 'users.createdAt', 'email', 'fullName')
+          .joinRelated('userDetails')
+          .castTo<UsersGetResponseDto>();
+      })
```

Comment:
The use of such amount of .withGraphJoined together with .modifyGraph (which forms subselect queries inside joins) calls can be avoided, but maybe the way how it could be accomplished won't look very pretty:
``` 
  /** @desc builded query could look like this */
      .query()
      .select(
        'interviews.*',
        raw(this.getFullRelatedObjectQuery('courseCategory')),
        raw(this.getMappedRelatedObjectQuery({
          relationFieldsMap: [
            { fieldName: 'id', fieldPath: '"interviewee"."id"' },
            { fieldName: 'createdAt', fieldPath: '"interviewee"."created_at"' },
            { fieldName: 'email', fieldPath: '"interviewee"."email"' },
            { fieldName: 'fullName', fieldPath: '"interviewee:user_details"."full_name"' },
          ],
          relationName: 'interviewee',
        })),
        raw(this.getMappedRelatedObjectQuery({
          relationFieldsMap: [
            { fieldName: 'id', fieldPath: '"interviewer"."id"' },
            { fieldName: 'createdAt', fieldPath: '"interviewer"."created_at"' },
            { fieldName: 'email', fieldPath: '"interviewer"."email"' },
            { fieldName: 'fullName', fieldPath: '"interviewer:user_details"."full_name"' },
          ],
          relationName: 'interviewer',
        })),
      )
      .joinRelated('courseCategory')
      .joinRelated('interviewee.userDetails')
      .joinRelated('interviewer.userDetails')
      .castTo<InterviewsGetAllItemResponseDto[]>()
      .execute();


  /** @desc just wraps joined table inside object */
  private getFullRelatedObjectQuery(relationName: string): string {
    const snakeCasedRelationName = changeStringCase({ caseType: StringCase.SNAKE_CASE, stringToChange: relationName });

    return `row_to_json("${snakeCasedRelationName}".*) AS "${relationName}"`;
  }

  /** @desc takes fields from defined path within joined tables and wraps them inside object */
  private getMappedRelatedObjectQuery({ relationFieldsMap, relationName }: {
    relationName: string,
    relationFieldsMap: Record<'fieldName' | 'fieldPath', string>[]
  }): string {
    const snakeCasedRelationName = changeStringCase({ caseType: StringCase.SNAKE_CASE, stringToChange: relationName });
    const fieldsFormation = relationFieldsMap.map(({ fieldName, fieldPath }) => {
			return `'${fieldName}', ${fieldPath}`;
		}).join(',');
    const nullishObjCondition = `THEN NULL ELSE JSONB_BUILD_OBJECT(${fieldsFormation})`;

    return `CASE WHEN "${snakeCasedRelationName}" IS NULL ${nullishObjCondition} END AS "${relationName}"`;
  }
```
That's not a call for a change - just a suggestion that may be discussed

Reply:
There's an approach by Alina, using `.modifiers()`, in #283, which, in my opinion, is easy to read and understand, and there are not a lot of subqueries
In my opinion, the one with `.modifiers()` is a better approach, but it's better to be discussed, to select one approach for all queries